src/main/java/dev/jbang/dependencies/DependencyAttributes.java

```diff
@@ -0,0 +1,29 @@
+package dev.jbang.dependencies;
+
+import java.util.*;
+
+public class DependencyAttributes {
+
+	private final Map<String, List<String>> attributes;
+
+	// Java 8 Set.of() is sad :)
+	static Set<String> defaultScopes = Collections.unmodifiableSet(new HashSet<>(Arrays.asList("build", "run")));
+	Set<String> scopes;
+
+	public DependencyAttributes(Map<String, List<String>> attributes) {
+		this.attributes = attributes;
+	}
+
+	// lazy computed; dont store hashset unless truly needed
+	public boolean includeInScope(String scope) {
```

Comment:
I don't like the word scope - mainly because these aren't actually maven scopes but rather something that gets translated into a scope.

`includeInConfig(String config)` maybe ? (gradle use configuration as name for different scenarios)

Reply:
and it would be `defaultConfigs` if follow same route.